alf/algorithms/mcts_algorithm.py

```diff
@@ -505,19 +646,59 @@ def _search(self, trees, to_plays):
         children_index = trees.children_index
         best_child_index = trees.best_child_index
         game_over = trees.game_over
+        psims = self._num_parallel_sims
+        parallel = self._parallel
         search_paths = []
         B = trees.B
         nodes = trees.root_indices
-        path_lengths = torch.ones_like(B)
+        calc_i = False
+        if parallel:
+            B = B.unsqueeze(-1)
+            nodes = trees.root_indices.unsqueeze(-1).expand(-1, psims)
+            if self._search_with_exploration_policy:
+                i = (torch.arange(psims), )
+            else:
+                calc_i = True
```

Comment:
In this case `i` is not initialized (I know this is not c++, but habits :) ) - Maybe we can just put the `i = ()` in the else branch right before the `if parallel` and remove the `else` branch below?

Reply:
Sounds good.